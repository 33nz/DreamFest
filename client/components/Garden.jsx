import React, { useState, useEffect, useContext } from 'react'
// import { UserContext } from './UserContext'

import Map from './Map'
import Events from './Events'

import { getUserGarden } from '../apiClient.js'

const Garden = (props) => {
  // const [user, setUser] = useContext(UserContext)
  const [name, setName] = useState(null)
  const [description, setDescription] = useState(null)
  const [url, setUrl] = useState(null)

  const gardenId = 1
  // const gardenId = user.garden_id

  useEffect(() => {
    getUserGarden(gardenId)
      .then(res => {
        setName(res.name)
        setDescription(res.description)
        setUrl(res.url)
        return null
      })
      .catch((error) => {
        console.log('error: ', error.message)
      })
  }, [])

  return (
    <>
      <div className="column">
        <h3>{name}</h3>
        <p className="mb-4">{description}</p>
        <a className="word-wrap" href={url}>{url}</a>
        <Events />
      </div>
      <Map/>
    </>
  )
}

export default Garden
