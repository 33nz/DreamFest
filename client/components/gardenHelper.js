import { dispatch, getState } from '../store'
import { setWaiting, clearWaiting } from '../actions/waiting'
import { showError } from '../actions/error'
import { getUserGarden } from '../api/gardens'

export function getGarden () {
  const storeState = getState()
  const { gardenId } = storeState.user
  dispatch(setWaiting())
  return getUserGarden(gardenId)
    .then(garden => {
      dispatch(clearWaiting())
      const { name, description, url, events, lat, lon } = garden
      return {
        name,
        description,
        url,
        events,
        lat,
        lon
      }
    })
    .catch((error) => {
      dispatch(showError(error.message))
    })
}
