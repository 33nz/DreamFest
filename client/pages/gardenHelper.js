import { dispatch, getState } from '../store'
import { setWaiting } from '../actions/waiting'
import { showError } from '../actions/error'
import requestor from '../consume'
import { setGarden } from '../actions/garden'

export function getGarden (consume = requestor) {
  const storeState = getState()
  const { gardenId } = storeState.user
  dispatch(setWaiting())
  return consume(`/gardens/${gardenId}`)
    .then((res) => {
      const { name, description, url, address, events, lat, lon } = res.body
      dispatch(setGarden({
        name,
        description,
        address,
        url,
        events,
        lat,
        lon
      }))
      return null
    })
    .catch((error) => {
      dispatch(showError(error.message))
    })
}
