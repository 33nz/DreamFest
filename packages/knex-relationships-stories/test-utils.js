const { within } = require('@testing-library/dom')
const { JSDOM } = require('jsdom')

const render = (response) => within(new JSDOM(response.text).window.document)

module.exports = { render }
