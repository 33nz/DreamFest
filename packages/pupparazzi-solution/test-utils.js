const { within } = require('@testing-library/dom')
const { JSDOM } = require('jsdom')

export const render = (response) =>
  within(new JSDOM(response.text).window.document)
