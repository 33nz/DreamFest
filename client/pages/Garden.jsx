import React, { useState, useEffect } from 'react'

import Map from '../components/Map'
import Events from '../components/Events'
import { getGarden, signedIn } from './gardenHelper'
import { Redirect } from 'react-router'

export default function Garden () {
  if (!signedIn()) return <Redirect to={'/signin'} />

  const [garden, setGarden] = useState({
    name: '',
    description: '',
    url: '',
    events: [],
    address: '',
    lat: 0,
    lon: 0
  })

  useEffect(() => {
    // eslint-disable-next-line promise/catch-or-return
    getGarden()
      .then((gardenData) => {
        setGarden(gardenData)
        return null
      })
  }, [])

  const { name, description, address, url, events, lat, lon } = garden
  return (
    <>
      <div className="column garden">
        <h2>{name}</h2>
        <p className="mb-4">{description} </p>
        <p><a className="gardenLink word-wrap" href={url}>{url}</a></p>
        <Events events={events} />
      </div>
      <Map
        coordinates={[{ lat: lat, lon: lon }]}
        addresses={[address]}
      />
    </>
  )
}
