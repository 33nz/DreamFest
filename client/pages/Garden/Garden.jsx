import React, { useEffect } from 'react'
import { useSelector } from 'react-redux'
import { useParams } from 'react-router-dom'

import Map from '../../components/Map/Map'
import Events from '../../components/events/Events/Events'
import { getGarden } from './gardenHelper'
import BarGraph from '../../components/dataVis/BarGraph'

export default function Garden () {
  const { id } = useParams()
  const garden = useSelector(globalState => globalState.garden)
  const user = useSelector(globalState => globalState.user)
  const location = useSelector(globalState => globalState.location)

  useEffect(() => {
    user.id && getGarden(id, user)
  }, [id, user])

  const { name, description, address, url, events, lat, lon } = garden

  return (
    <section className='flex-container'>
      <div className='column-6'>
        <article className='column-9 garden'>
          <h2>{name}</h2>
          <p>{description}</p>
          <a href={url}>{url}</a>
        </article>
        <Events gardenid={id} events={events} />
      </div>
      <section className='graph-map'>
        <Map
          userCoordinates={location}
          coordinates={[{ lat: lat, lon: lon }]}
          addresses={[address]}
          names={[name]}
        />
        {user.isAdmin ? <BarGraph events={events}/> : null}
      </section>
    </section>
  )
}
