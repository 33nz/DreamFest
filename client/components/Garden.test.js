import React from 'react'
import { screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Garden from './Garden'
import { getGarden } from './gardenHelper'
import { renderWithRedux } from '../test-utils'

jest.mock('./gardenHelper')

test('calls getGarden and renders garden data on mount', () => {
  getGarden.mockImplementation(() => Promise.resolve({
    name: 'test garden',
    description: 'an excellent test garden',
    url: 'cooltestgarden.com',
    events: []
  }))
  renderWithRedux(<Garden />)
  return screen.findByRole('heading', { name: 'test garden' })
    .then((title) => {
      const url = screen.getByRole('link', { name: 'cooltestgarden.com' })
      const description = screen.getByText((text) => text.includes('excellent'))
      expect(title).toHaveTextContent('test garden')
      expect(url).toHaveTextContent('cooltestgarden.com')
      expect(description).toHaveTextContent(/test garden$/)
      expect(url.href).toMatch('cooltestgarden.com')
      return null
    })
})
