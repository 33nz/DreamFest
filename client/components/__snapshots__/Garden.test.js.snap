// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="column"
  >
    <h3>
      Kelmarna Gardens
    </h3>
    <p
      className="mb-4"
    >
      Kelmarna Gardens is a city farm and organic community garden, situated on 4.5 acres of council land in Ponsonby, close to the heart of Auckland City.
    </p>
    <a
      className="word-wrap"
      href="http://www.kelmarnagardens.nz/"
    >
      http://www.kelmarnagardens.nz
    </a>
    <h2
      className="my-4"
    >
      Events
    </h2>
    <div
      className="box"
    >
      <div>
        <h4>
          Weeds worker bee
        </h4>
        <p>
          28/09/2020 8.00 am
        </p>
        <p>
          8
           volunteers needed
        </p>
        <p>
          We need to weed the upper planting beds
        </p>
      </div>
    </div>
  </div>,
  <div
    className="column"
  >
    <img
      src="/placeholder_auckland.jpg"
    />
    <p>
      12 Hukanui Crescent
    </p>
  </div>,
]
`;

exports[`renders correctly 2`] = `
Array [
  <div
    className="column"
  >
    <h3>
      Kelmarna Gardens
    </h3>
    <p
      className="mb-4"
    >
      Kelmarna Gardens is a city farm and organic community garden, situated on 4.5 acres of council land in Ponsonby, close to the heart of Auckland City.
    </p>
    <a
      className="word-wrap"
      href="http://www.kelmarnagardens.nz/"
    >
      http://www.kelmarnagardens.nz
    </a>
    <h2
      className="my-4"
    >
      Events
    </h2>
    <div
      className="box"
    >
      <div>
        <h4>
          Weeds worker bee
        </h4>
        <p>
          28/09/2020 8.00 am
        </p>
        <p>
          8
           volunteers needed
        </p>
        <p>
          We need to weed the upper planting beds
        </p>
      </div>
    </div>
  </div>,
  <div
    className="column"
  >
    <img
      src="/placeholder_auckland.jpg"
    />
    <p>
      12 Hukanui Crescent
    </p>
  </div>,
]
`;

exports[`renders correctly 3`] = `
Array [
  <div
    className="column"
  >
    <h3>
      Kelmarna Gardens
    </h3>
    <p
      className="mb-4"
    >
      Kelmarna Gardens is a city farm and organic community garden, situated on 4.5 acres of council land in Ponsonby, close to the heart of Auckland City.
    </p>
    <a
      className="word-wrap"
      href="http://www.kelmarnagardens.nz/"
    >
      http://www.kelmarnagardens.nz
    </a>
    <h2
      className="my-4"
    >
      Events
    </h2>
    <div
      className="box"
    >
      <div>
        <h4>
          Weeds worker bee
        </h4>
        <p>
          28/09/2020 8.00 am
        </p>
        <p>
          8
           volunteers needed
        </p>
        <p>
          We need to weed the upper planting beds
        </p>
      </div>
    </div>
  </div>,
  <div
    className="column"
  >
    <img
      src="/placeholder_auckland.jpg"
    />
    <p>
      12 Hukanui Crescent
    </p>
  </div>,
]
`;
